src/antq/core.clj

```diff
@@ -101,10 +101,40 @@
    [nil "--changes-in-table"]
    [nil "--transitive"]])
 
+(defn- retrieve-artifact-name
+  "Retrieve artifact name from artifact string, which may contain version after `@` sign"
+  [artifact]
+  (let [specific-version (str/split (str artifact) #"@" 2)]
+    (if (some? (second specific-version)) (first specific-version) artifact)))
+
+(defn forced-artifacts
+  "Forced artifacts are coming from focus param and contain specific version targeted with @"
+  [options]
+  (->> (:focus options)
+       (remove (fn [artifact]
+                 (-> (str/split (str artifact) #"@" 2)
+                     second
+                     not)))
+       (map (fn [art]
+              (let [[name ver] (str/split (str art) #"@" 2)]
+                {:name name, :latest-version ver})))))
+
+(defn forced-version
+  "Returns forced version if exists for `dep`, otherwise nil"
+  [dep forced-artifacts]
+  (when-let [matching-artifact (some (fn [artifact]
+                                       (when (= (:name artifact) (:name dep))
+                                         artifact))
+                                     forced-artifacts)]
+    (:latest-version matching-artifact)))
```

Comment:
When using the result of `forced-artifact-version-map`, you can simply use `(get % (:name dep))`.